pkg/apis/config/feature_flags.go

```diff
@@ -337,6 +342,12 @@ func CheckWarnResourceVerificationMode(ctx context.Context) bool {
 	return cfg.FeatureFlags.ResourceVerificationMode == WarnResourceVerificationMode
 }
 
+// CheckAlphaOrBetaAPIFields return true if the enable-api-fields is either set to alpha or set to beta
+func CheckAlphaOrBetaAPIFields(ctx context.Context) bool {
+	cfg := FromContextOrDefaults(ctx)
+	return cfg.FeatureFlags.EnableAPIFields == AlphaAPIFields || cfg.FeatureFlags.EnableAPIFields == BetaAPIFields
+}
+
```

Comment:
not related to this PR -- @chitrangpatel what do you think of cleaning up propagated workspaces to reuse this function? just to make the logic clearer?

Reply:
Yes, can do! I think it's best to stay consistent. Once this is merged, I will open an issue and implement a fix.

Reply:
sounds good, thanks @chitrangpatel!